pkgdb/src/buildenv/realise.cc

```diff
@@ -60,36 +60,171 @@ namespace flox::buildenv {
 #  error "COMMON_NIXPKGS_URL must be set to a locked flakeref of nixpkgs to use"
 #endif
 
-#ifndef FLOX_BASH_BIN
-#  error "FLOX_BASH_BIN must be set to the path of a Bash executable"
+#ifndef FLOX_BASH_PKG
+#  error "FLOX_BASH_PKG must be set to the path of the nix bash package"
+#endif
+
+#ifndef FLOX_COREUTILS_PKG
+#  error \
+    "FLOX_COREUTILS_PKG must be set to the path of the nix coreutils package"
+#endif
+
+#ifndef FLOX_GNUSED_PKG
+#  error "FLOX_GNUSED_PKG must be set to the path of the nix gnused package"
+#endif
+
+#ifndef FLOX_PROCPS_PKG
+#  error "FLOX_PROCPS_PKG must be set to the path of the nix procps package"
 #endif
 
 /* -------------------------------------------------------------------------- */
 
+// Top-level activate script, always invoked with nix bash.
+const char * const ACTIVATE_SCRIPT = R"_(
+# Flox environment activation script.
+[ ${_FLOX_PKGDB_VERBOSITY:-0} -eq 0 ] || set -x
+
+# Capture starting environment.
+_start_env="$($_coreutils/bin/mktemp --suffix=.start-env)"
+_end_env="$($_coreutils/bin/mktemp --suffix=.end-env)"
+export | $_coreutils/bin/sort > "$_start_env"
+
+function cleanup() {
+  $_coreutils/bin/rm -f "$_start_env" "$_end_env"
+}
+
+# Set FLOX_ENV as the path by which all flox scripts can make reference to
+# the environment to which they belong. Use this to define the path to the
+# activation scripts directory.
+FLOX_ENV="$( $_coreutils/bin/dirname -- "${BASH_SOURCE[0]}" )"
+export FLOX_ENV
+
+# Process the flox environment customizations, which includes (amongst
+# other things) prepending this environment's bin directory to the PATH.
+if [ -d "$FLOX_ENV/etc/profile.d" ]; then
+  declare -a _prof_scripts;
+  _prof_scripts=( $(
+    cd "$FLOX_ENV/etc/profile.d";
+    shopt -s nullglob;
+    echo *.sh;
+  ) );
+  for p in "${_prof_scripts[@]}"; do . "$FLOX_ENV/etc/profile.d/$p"; done
+  unset _prof_scripts;
+fi
+
+# Set static environment variables from the manifest.
+if [ -f "$FLOX_ENV/activate.d/envrc" ]; then
+  source "$FLOX_ENV/activate.d/envrc"
+fi
+
+# Source the hook-on-activate script if it exists.
+if [ -e "$FLOX_ENV/activate.d/hook-on-activate" ]; then
+  # Nothing good can come from output printed to stdout in the
+  # user-provided hook scripts because these can get interpreted
+  # as configuration statements by the "in-place" activation
+  # mode. So, we'll redirect stdout to stderr.
+  source "$FLOX_ENV/activate.d/hook-on-activate" 1>&2
+elif [ -e "$FLOX_ENV/activate.d/hook-script" ]; then
+  # [hook.script] is deprecated - print warning and source it.
+  echo "WARNING: [hook.script] is deprecated, use [hook.on-activate] instead." >&2
```

Comment:
Ah, thanks for mentioning, didn't know we were waiting for that. By the time this is merged (hopefully next release) will that have been long enough to enable the warning?

The exact equivalent is:
- `hook.on-activate` because that is the only place you can place things that a) run only once and b) the effects of which are inherited by all subshells
- _and_ `profile.common` for invoking commands to be invoked with all subshells

(different shells have different ways to define aliases so those are better placed in the shell-specific sections)

We could update the warning to say "WARNING: [hook.script] is deprecated; use [hook.on-activate] for commands run once per activation and [profile.common] for commands which run in all sub-shells". It's a bit long - what do you think?

Reply:
Have since removed this section of code; as you mentioned we weren't planning to warn about it just yet.